### PR TITLE
test(cli): exercise interactive selector routes

### PR DESCRIPTION
**Pull request by Sinity**
## Summary

Add a real PTY test boundary for the interactive CLI: Click shell completion, fzf selection, and terminal-size propagation.

## Problem

Existing terminal tests captured output but did not attach a controlling terminal, invoke the real external picker, or execute Click shell completion through its installed process protocol. Those gaps could let interactive-only regressions pass.

## Solution

The PTY harness now applies its requested window size and gives the child process a controlling terminal. Three process-route tests use an isolated archive and deterministic fzf stand-in to prove completion output, ranked picker selection, and picker-visible terminal dimensions. The child-width change also refreshes the affected terminal snapshots; that full fixture refresh records pre-existing annotations-command inventory drift at each tested width.

## Verification

- env POLYLOGUE_PYTEST_BASETEMP_ROOT=/realm/tmp/polylogue-pytest POLYLOGUE_PYTEST_WORKERS=1 nix develop --command devtools test tests/unit/cli/test_terminal_snapshots.py tests/unit/cli/test_interactive_cli.py
  - 12 passed in 35.64s
- Pre-push quick verification passed before each pushed commit.

## Acceptance criteria

Ref polylogue-th0

- Deferred to polylogue-th0: the five-plus golden-flow serial CI lane is not implemented in this PR.
- Deferred to polylogue-th0: this PR does not add the required registry-generated completion contract; the existing manually curated completion matrix does not satisfy that AC.
- Deferred to polylogue-th0: the first-row fzf test makes candidate-order regression observable, but this branch has no retained deliberate mutation-demonstration commit as required by the bead.

## Anti-vacuity

- test_click_bash_completion_protocol_runs_in_a_real_pty executes the CLI entrypoint and Click shell-complete dispatch. Removing root --origin completion or changing the bash completion records makes it fail.
- test_select_uses_fzf_first_ranked_candidate_in_a_real_pty executes query routing, selection, and the external fzf process. Skipping fzf, ignoring its result, or reordering candidates makes it fail.
- test_select_propagates_requested_pty_size_to_fzf executes the same selector route and reads its controlling terminal from fzf. Removing PTY size or controlling-terminal setup makes it fail.

## Adversarial review notes

- Iteration 1: refreshed terminal snapshots after the real PTY width became visible to the CLI, in 0ec19eb5a. Strengthened the completion oracle from a rendered substring to the exact Click bash protocol records in the same commit.
- Iteration 2: corrected the remaining fzf acceptance claim. The test is a real route and exposes reordering, but the deliberate mutation demonstration is not present in this branch and remains deferred to polylogue-th0.
- Iteration 3: no legitimate gaps found. The PR body and review-thread state accurately distinguish this shipped test slice from the open bead requirements.
- The five-flow serial lane and registry-generated completion matrix remain deferred to open bead polylogue-th0.

### CODE REVIEW

**Comment by Sinity** [comment]
Adversarial self-review iteration 1

Findings:
- Real: applying real PTY dimensions invalidated three stored terminal snapshots. Fixed in 0ec19eb5a by regenerating the affected snapshot fixture.
- Real: the PR had overstated the existing completion matrix as registry-generated coverage. The PR body now states that this AC remains deferred to open bead polylogue-th0.
- Borderline strengthened: the Click completion route test now asserts the exact bash completion records, not a rendered substring.
- Open scope retained honestly: the five-plus-flow serial CI lane and the registry-generated completion contract remain deferred to polylogue-th0.

Verification: env POLYLOGUE_PYTEST_BASETEMP_ROOT=/realm/tmp/polylogue-pytest POLYLOGUE_PYTEST_WORKERS=1 nix develop --command devtools test tests/unit/cli/test_terminal_snapshots.py tests/unit/cli/test_interactive_cli.py passed 12 tests.

What survived scrutiny: the fzf route is a real CLI to external-process path; its deterministic first-row chooser still fails if candidate order or the selected result changes.

File: tests/infra/pty_cli.py [hunk 1]
```diff
@@ -130,8 +133,16 @@ def run_in_pty(
 
     # Open PTY
     master_fd, slave_fd = os.openpty()
+    # ``openpty`` does not inherit the requested dimensions.  Set them before
+    # spawning so Rich, fzf, and any future full-screen route see the same
+    # terminal shape that pyte renders for assertions.
+    fcntl.ioctl(slave_fd, termios.TIOCSWINSZ, struct.pack("HHHH", rows, cols, 0, 0))
```
[the remark this answers is not in the document]
Comment (Sinity) [reply tests/infra/pty_cli.py:139]: Addressed in 0ec19eb5a: regenerated tests/unit/cli/__snapshots__/test_terminal_snapshots.ambr after TIOCSWINSZ changed child layout. Focused terminal snapshots plus interactive routes pass: 12 tests.

**Comment by Sinity** [comment]
Adversarial self-review iteration 2

Findings:
- Real: the bead requires a deliberate candidate-reordering demonstration commit. This branch has a real fzf route test that would fail on reordered input, but no retained mutation demonstration, so the PR now defers this AC to polylogue-th0 instead of calling it satisfied.
- Borderline documented: snapshot regeneration also captured pre-existing annotations-command inventory drift at each tested width; no production behavior changed.

The existing inline snapshot review thread is resolved with the 12-test focused verification evidence. No code changed in this iteration, so no additional commit was appropriate.

What survived scrutiny: the fzf selection and terminal-size paths remain real subprocess/TTY routes, and the completion assertion now checks Clicks exact bash protocol records.

**Comment by Sinity** [comment]
Adversarial self-review iteration 3

No new legitimate gaps found. The full branch diff, bead notes and ACs, PR conversation, and thread-aware review state were rechecked. The only substantive inline review is resolved; CodeRabbit remains rate-limited, and the known all-jobs-at-init CI failure pattern was not treated as a branch finding.

What survived scrutiny: the completion, fzf selection, and terminal-size tests exercise real child processes and a controlling TTY. The three larger bead requirements remain accurately deferred rather than implied complete.

**Comment by Sinity** [comment]
Final acceptance-criteria matrix for polylogue-th0

| Acceptance criterion | Status | Evidence / disposition |
| --- | --- | --- |
| 5+ PTY golden flows in a serial CI lane | Deferred to bead | This PR ships three real-route exemplars only; the serial CI lane and remaining flows stay in open bead polylogue-th0. |
| Registry-generated completion contracts that reject missing metadata | Deferred to bead | The existing matrix is manually curated; this PR does not claim otherwise. The registry-generated contract remains open scope in polylogue-th0. |
| Deliberate candidate-reordering regression caught in a demonstration commit | Deferred to bead | The first-row fzf test makes reordering observable, but this branch has no retained mutation-demonstration commit. |

No criterion is marked satisfied in full. This PRs delivered slice is the PTY substrate, three non-vacuous real-process exemplars, exact bash-completion protocol coverage, and refreshed terminal-width snapshots.
